### PR TITLE
Zero thickness for volumetric equivalence of 2D material

### PR DESCRIPTION
**Pull request by weiliangjin2021**
Different subclass for volumeric equivalence of Medium2D

### CODE REVIEW

File: tidy3d/components/medium.py [hunk 1]
```diff
@@ -4287,6 +4287,12 @@ def sel_inside(self, bounds: Bound):
         return self.updated_copy(**dict(zip(["xx", "yy", "zz"], new_comps)))
 
 
+class AnisotropicMediumFromMedium2D(AnisotropicMedium):
+    """The same as ``AnisotropoicMeidum``, but converted from Medium2D.
```
Comment (dmarek-flex) [inline tidy3d/components/medium.py]: typo "Anisotropic Medium"

**Comment by dmarek-flex** [comment]
Looks good, but I think I will wait to approve until I try it with the backend changes.

File: tidy3d/components/simulation.py [hunk 2]
```diff
@@ -3638,10 +3638,7 @@ def snap_to_grid(geom: Geometry, axis: Axis) -> Geometry:
                 temp_geometry = set_bounds(snapped_geometry, bounds=new_bounds, axis=axis)
```
Comment (weiliangjin2021) [inline tidy3d/components/simulation.py:3638]: I'm not sure about this line (and the line above):
```
new_bounds = (snapped_center - dls[0] / 2, snapped_center + dls[1] / 2)
temp_geometry = set_bounds(snapped_geometry, bounds=new_bounds, axis=axis)
```
The bound here is still of dl-thickness. Is that the right approach @dmarek-flex ?

Comment (dmarek-flex) [reply tidy3d/components/simulation.py]: Good point.
Now that we have special treatment for 2d materials I don't think we need to include the volumetric equivalent in the list of structures that is passed to the `subdivide` function. So we don't need the `temp_geometry` or `background_structures_temp`. new_bounds should start as (center,center)

**Review by tylerflex — APPROVED** [review]
Looks good. just a few comments
* Do we want to allow `AnisotropicMediumFromMedium2D` to be imported form `tidy3d`? if so, import it into `__init__.py`.
* Please add changelog line, if you think it warrants it.
Thanks!

**Review by dmarek-flex — APPROVED** [review]

**Review by caseyflex — APPROVED** [review]

**Pull request merged**